libraries/node-core-library/src/Import.ts

```diff
@@ -83,4 +128,105 @@ export class Import {
     const importLazyLocal: (moduleName: string) => unknown = importLazy(require);
     return importLazyLocal(moduleName);
   }
+
+  /**
+   * Resolves a path in a package, relative to another path.
+   */
+  public static resolve(options: IImportResolveOptions): string {
+    const { resolvePath } = options;
+
+    if (nodeJsPath.isAbsolute(resolvePath)) {
+      return resolvePath;
+    }
+
+    let normalizedRootPath: string = FileSystem.getRealPath(options.baseFolderPath);
```

Comment:
It seems like `require.sync()` already implements all these parsing features itself.  It almost exactly simulates Node.js `require()` behavior.